### PR TITLE
Added debug mode for testing with debuggers

### PR DESCRIPTION
**Pull request by BelfordZ**
**Debugging tests via DevTools**

- Added debug mode for testing with debuggers
- Document using test:debug (with references)

**Description**

- ADDED:
  - debug mode for testing with debuggers
    - We need to be able to easily step through the code with DevTools while running tests so that we can reasonably begin to address #493 (or any complex controller interaction for that matter).

**Checklist**

- [ x ] Tests are included if applicable
- [ x ] Any added code is fully documented

Thx @shanejonas for the nudge on this one :)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,6 +32,8 @@
     "prepack": "./scripts/prepack.sh",
     "setup": "yarn install",
     "test": "jest",
+    "test:debug": "node --inspect-brk node_modules/.bin/jest --runInBand",
+    "test:debug:windows": "node --inspect-brk ./node_modules/jest/bin/jest.js --runInBand",
```
Comment (mcmire) [inline package.json:36]: I feel like we should minimize the number of scripts we have, so is there a way that we can look for the location of `jest`? Perhaps we can extract this to a script in `scripts/` and say something like:

``` sh
#!/bin/sh

exec node --inspect-brk $(yarn bin jest) --runInBand
```

That way we only need one script.

Comment (BelfordZ) [reply package.json:36]: nice one! what if we just put that straight in the package.json script for `test:debug` ?

```suggestion
    "test:debug": "node --inspect-brk $(yarn bin jest) --runInBand",
```

Comment (Gudahtt) [reply package.json]: This wouldn't work on Windows though. `cmd` doesn't support the `$( .. )` syntax.

I'd recommend using [`yarn run`](https://yarnpkg.com/cli/run#options), it supports the `--inspect-brk` command.

e.g.

```
"test:debug": "yarn run --inspect-brk jest --runInBand",
```

Comment (BelfordZ) [reply package.json]: @Gudahtt ooooooo nice catch!

File: package.json [hunk 2]
```diff
@@ -32,6 +32,7 @@
     "prepack": "./scripts/prepack.sh",
     "setup": "yarn install",
     "test": "jest",
+    "test:debug": "node --inspect-brk $(yarn bin jest) --runInBand",
```
Comment (BelfordZ) [inline package.json:35]: TY @Gudahtt 

```suggestion
    "test:debug": "yarn run --inspect-brk jest --runInBand",
```

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**